src/s3ql/backends/s3c.py

```diff
@@ -547,7 +547,7 @@ def _parse_error_response(self, resp):
             raise HTTPError(resp.status, resp.reason, resp.headers)
 
         # If not XML, do the best we can
-        if not XML_CONTENT_RE.match(content_type) or resp.length == 0:
+        if not (isinstance(content_type, str) and XML_CONTENT_RE.match(content_type)) or resp.length == 0:
```

Comment:
This does not look right to me. What other (non-str) type are we expecting here?

Reply:
We are guarding against any non-string types, including nil. See issue #278.

Reply:
Ah, in that case please check against this explicitly - `if content_type is None or resp.length == 0 or XML_CONTENT_RE.match(content_type)`.